packages/amazonq/src/lsp/client.ts

```diff
@@ -160,120 +160,141 @@ export async function startLanguageServer(
 
     const disposable = client.start()
     toDispose.push(disposable)
+    await client.onReady()
 
-    const auth = new AmazonQLspAuth(client)
+    const auth = await initializeAuth(client)
 
-    return client.onReady().then(async () => {
-        await auth.refreshConnection()
+    await postStartLanguageServer(auth, client, resourcePaths, toDispose)
 
-        if (Experiments.instance.get('amazonqLSPInline', false)) {
-            const inlineManager = new InlineCompletionManager(client)
-            inlineManager.registerInlineCompletion()
-            toDispose.push(
-                inlineManager,
-                Commands.register({ id: 'aws.amazonq.invokeInlineCompletion', autoconnect: true }, async () => {
-                    await vscode.commands.executeCommand('editor.action.inlineSuggest.trigger')
-                }),
-                vscode.workspace.onDidCloseTextDocument(async () => {
-                    await vscode.commands.executeCommand('aws.amazonq.rejectCodeSuggestion')
-                })
-            )
-        }
+    return client
+}
 
-        if (Experiments.instance.get('amazonqChatLSP', true)) {
-            await activate(client, encryptionKey, resourcePaths.ui)
-        }
+async function initializeAuth(client: LanguageClient): Promise<AmazonQLspAuth> {
+    AuthUtil.instance.regionProfileManager.onDidChangeRegionProfile(async () => {
```

Comment:
technically this listener could trigger before we push the bearer token below in `await auth.refreshConnection(true)` right? Wouldn't that be problematic?

Reply:
From testing I haven't found a code path where this happens